imnodes.cpp

```diff
@@ -1834,6 +1871,17 @@ void Link(int id, const int start_attr_id, const int end_attr_id)
     link.color_style.base = g.style.colors[ColorStyle_Link];
     link.color_style.hovered = g.style.colors[ColorStyle_LinkHovered];
     link.color_style.selected = g.style.colors[ColorStyle_LinkSelected];
+
+    // Check if this link was created by the current link event
+    if (editor.click_interaction_type == ClickInteractionType_LinkCreation &&
+        editor.pins.pool[link.end_pin_idx].flags & AttributeFlags_EnableLinkCreationOnSnap &&
+       (editor.click_interaction_state.link_creation.start_pin_idx == link.start_pin_idx &&
+        editor.click_interaction_state.link_creation.end_pin_idx == link.end_pin_idx) ||
+       (editor.click_interaction_state.link_creation.start_pin_idx == link.end_pin_idx &&
+        editor.click_interaction_state.link_creation.end_pin_idx == link.start_pin_idx))
```

Comment:
Woops, had a comment for this in the review, but not sure where it went.

We should wrap the LHS of the `||` expression in parenthesis to fix the `logical-op-parenthesis` warning:
```
imnodes.cpp
../../imnodes.cpp:1880:76: warning: '&&' within '||' [-Wlogical-op-parentheses]
    if (editor.click_interaction_type == ClickInteractionType_LinkCreation &&
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~
```

Agreed on the jankiness of this addition, but offhand, can't really think of anything better either 😄 My gut reaction was to move this to the link creation handling part of `IsLinkCreated`, but we can't since we need an id! But thankfully it's an implementation detail, it can be changed if a better idea pops up 😛

Reply:
Fixed